reader/xml/decoder_test.go

```diff
@@ -0,0 +1,35 @@
+package xml // import "miniflux.app/reader/xml"
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+	"testing"
+
+	"miniflux.app/reader/encoding"
+)
+
+func Test(t *testing.T) {
+	type myxml struct {
+		XMLName xml.Name `xml:"rss"`
+		Version string   `xml:"version,attr"`
+		Title   string   `xml:"title"`
+	}
+	// Add the body contains illegal characters
+	data := fmt.Sprintf(`<?xml version="1.0" encoding="windows-1251"?><rss version="2.0"><title>%s</title></rss>`, "\x10")
+	var x myxml
+	decoder := xml.NewDecoder(strings.NewReader(data))
```

Comment:
You don't need to test the default XML decoder, just the new one below is enough.

Reply:
It's more a demonstration. Removed